app/models/user.rb

```diff
@@ -70,7 +78,23 @@ def password_reset_expired?
   end
 
   def feed
-    Micropost.find_by_user id
+    Micropost.get_feed(id).or(Micropost.find_by_user(id))
+  end
+
+  def follow(other_user)
+    if !following?(other_user)
+      following << other_user
```

Comment:
phần xử lý này là như nào em nhỉ?

Reply:
Sau khi được liên kết nhiều-nhiều (qua `has_many`,...) ở phía trên, ta nhận được một thuộc tính `following` có hành vi rất giống `array`. Ví dụ, ta có thể kiểm tra xem trong danh sách các user được follow có bao gồm một người dùng nào đó hay không bằng:
```ruby
following.include? some_user
```
Ta cũng có thể thêm hay xóa phần tử như thể nó là `array`:
```ruby
following.push some_user
following.delete some_user
```
Có thể thay `push` bằng một toán tử `<<`, gọi là *shovel operator*:
```ruby
following << some_user
```
Vậy phần trên đơn giản chỉ là: nếu check thấy `other_user` chưa được follow, thì ta đưa nó vào danh sách following và lưu vào CSDL.